vault/plugin_reload.go

```diff
@@ -27,18 +27,28 @@ func (c *Core) reloadMatchingPluginMounts(ctx context.Context, mounts []string)
 
 	var errors error
 	for _, mount := range mounts {
+		var isAuth bool
+		// allow any of
+		//   - sys/auth/foo/
+		//   - sys/auth/foo
+		//   - auth/foo/
+		//   - auth/foo
+		if strings.HasPrefix(mount, credentialRoutePrefix) {
+			isAuth = true
+		} else if strings.HasPrefix(mount, systemMountPath+credentialRoutePrefix) {
+			isAuth = true
+			mount = strings.TrimPrefix(mount, systemMountPath)
+		}
+		if !strings.HasSuffix(mount, "/") {
+			mount += "/"
+		}
+
 		entry := c.router.MatchingMountEntry(ctx, mount)
 		if entry == nil {
 			errors = multierror.Append(errors, fmt.Errorf("cannot fetch mount entry on %q", mount))
 			continue
 		}
 
-		var isAuth bool
-		fullPath := c.router.MatchingMount(ctx, mount)
-		if strings.HasPrefix(fullPath, credentialRoutePrefix) {
-			isAuth = true
-		}
-
 		// We dont reload mounts that are not in the same namespace
 		if ns.ID != entry.Namespace().ID {
 			continue
```

Comment:
This looks like it would also be returning a fake success here when in fact the reload fails. Would changing this too also be an improvement?

Reply:
Good thinking! I wasn't sure so I tested it and we get the following error when using namespaces
```
Error reloading plugin/mounts: Error making API request.

Namespace: ns2/
URL: PUT http://127.0.0.1:8200/v1/sys/plugins/reload/backend
Code: 500. Errors:

* 3 errors occurred:
	* cannot fetch mount entry on "auth/userpass-single-sdk-1_9-1/"
	* cannot fetch mount entry on "auth/userpass-single-sdk-1_9-2/"
	* cannot fetch mount entry on "auth/userpass-single-sdk-1_9-3/"
```

The call to [c.router.MatchingMountEntry](https://github.com/hashicorp/vault/pull/15579/files#diff-e73eb5b8e9c4fd6c530672fefd33a171454a1f0ebf6964f9cd3242f6aef88da1R46) above will ensure the namespace is applied before searching for the mount. So I think this check may be redundant?